rest_server/RestServer.go

```diff
@@ -0,0 +1,269 @@
+package main
+
+// specs from https://github.com/siodb/siodb/blob/master/docs/dev/designs/RestServer.md
+// TODO 1) Create the directory with code and give you build instruction so that you can integrate it (even if not 100% compliant yet)
+//      2) Finish the POST and DELETE implementation
+//      3) Integrate the rest of the requirements you mentioned (log format, etc)
+// TODO: Put chunked WritePayload in function
+// TODO: Put read chunked payload from IOMgr in function
+// TODO: Once cxxman code step 5 in https://github.com/siodb/siodb/blob/master/docs/dev/designs/RestServer.md#post-patch-requests
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"log"
+	"os"
+	"regexp"
+	"strconv"
+	"strings"
+)
+
+type RestServer struct {
+	Ipv4HTTPPort          uint64
+	Ipv4HTTPSPort         uint64
+	Ipv6HTTPPort          uint64
+	Ipv6HTTPSPort         uint64
+	TLSCertificate        string
+	TLSPrivateKey         string
+	HTTPChunkSize         uint64
+	logFileDestination    string
```

Comment:
that's wrong. example log channel names provided in the config file are not fixed, I could name them any other name and there can be more than 2 log channels.
list of log channels is in the paramerter "log_channels = file, console"
assume your code must handle correctly something like this:
```
log_channels = ch1, ch2, ch3, ch4, ch5, ch6, ch7, ch8, ch9, ch10
...
log.ch1.type=console
log.ch2.type=file
...
```
I suggest to use `map[string] LogChannelOptions`, where `LogChannelOptions` is structure with log channel options. See Siodb code for similar one: https://github.com/siodb/siodb/blob/d2ec5cfad8ecf727a9de0b69e73257a6662594c7/common/lib/siodb/common/options/LogOptions.h#L41